source/resampling_with_code2.Rmd

```diff
@@ -497,10 +501,19 @@ for item in ['shoe', 'saddle', 'jacket']:
     print('I am stitching a', item)
 ```
 
+```{r}
+items <- c('shoe', 'saddle', 'jacket')
+for (item in items) {
+  message('I am stitching a ', item)
+}
+```
+
 Here, the robot is working its way through three items: a shoe, a saddle, and a jacket.
-As it moves on to each object, it assigns the label ("shoe", "saddle", etc.) to the variable `item`.
-The indented block of code under the for is executed for each item, and in this code block `label` can be accessed.
-The code above can therefore be unpacked into the following equivalent program:
+Every time that a new object arrives in front of it, that object is assigned the label `item`.
+The [indented block of code under]{.python}[block of code inside]{.r} the `for` is executed for each one, and in this code block `label` can be accessed.
+You will also notice us using a new function: [`print`]{.python}[`message`]{.r}. This function receives several arguments, and then prints them out to screen.
```

Comment:
Why `message` and not `print` for R?   A little research - https://www.geeksforgeeks.org/printing-output-of-an-r-program - suggests that `message` is a bit unusual.

Reply:
Because `print` requires `print(paste(...))`.

Reply:
Yes, that's correct, you cannot do `print("Hello", x)` in R. :man_shrugging:

Reply:
Hmm - yes - ugly - I agree then, that message makes more sense.  But then - I guess we should be using it everywhere.

Reply:
I am no R expert, so I fully defer to you on what to use in that world.

Reply:
I think it's OK to stick to message for this - I haven't seen it before, but I see the problem here.   I just mean that we should be consistent.